migrations/src/postDeploymentSetup.js

```diff
@@ -44,22 +43,19 @@ module.exports = async () => {
   // Only transfer tokens in testnets
   if (NETWORK !== 'mainnet') {
     // Add new instance of StakeManager contract & Deployer address as Minter
-    await schellingCoin.addMinter(stakeManagerAddress);
+    const initialSupply = await schellingCoin.INITIAL_SUPPLY();
     await schellingCoin.addMinter(signers[0].address);
 
     if (SCHELLING_COIN_ADDRESS !== '') {
-      const { StakeManager: oldStakeManagerAddress } = await readOldDeploymentFile();
-
       // if previous instances of Schelling Coin is reused again and again,
       // then initial balance will get depleted, thus intial tokens minting is needed,
       // each time Schelling Coin instance is reused
       const initialSupply = await schellingCoin.INITIAL_SUPPLY();
       await schellingCoin.mint(signers[0].address, initialSupply);
 
       // Remove previous instance of StakeManager contract & Deployer address from Minter
-      await schellingCoin.removeMinter(oldStakeManagerAddress);
-      await schellingCoin.removeMinter(signers[0].address);
     }
+    await schellingCoin.mint(stakeManagerAddress, initialSupply);
```

Comment:
Hey Hi @adi44,
If I understand this PR correctly 
initialSupply is minted twice once at line no 54 to signers[0] and once here.
The thing is we have diff figure for mintable supply in our tokenomics,
Please mint that only to stakeManager.
Also only mint InitialSupply- MintableSupply to signers[0]